tests/TemporaryDirectoryTest.php

```diff
@@ -293,6 +293,20 @@ public function it_exists_function_should_tell_if_directory_exists()
         $this->assertTrue($temporaryDirectory->exists());
     }
 
+    /** @test */
+    public function it_can_delete_when_object_is_destroyed()
+    {
+        $temporaryDirectory = (new TemporaryDirectory())
+            ->name($this->temporaryDirectory)
+            ->deleteWhenDestroyed()
+            ->create();
+
+        $this->assertDirectoryExists($fullPath = $temporaryDirectory->path());
```

Comment:
That's interesting. I didn't know PHP could assign while passing an argument. I have to admit it confused me at first, though, and I'll leave it to @spatie to decide if it's a desirable practice.